### DISCUSSION TITLE
Show package downloads by version over time

### DISCUSSION THREAD

**Discussion by florian-richter** [community/community#128408] [npm] [answered]
NPM introduced the download number of package by version as requested in https://github.com/npm/feedback/discussions/60 and released in https://github.com/npm/roadmap/issues/19.

As a developer of several public NPM packages, I would love to see how adaption evolves over time. This can probably be achieved by scraping the website every week, but it would be preferable if the download by version number would be available via an API or an interface on npmjs.org. Seeing how downloads shift across versions would allow us to understand adoption. Especially it would make it easier to map events (like conference presentations or packages that use us as a dependency updating) to changes in adoption.

**Comment by monishcm** [comment, marked as answer]
Thanks @florian-richter for the feature request and we are considering to implement a solution that addresses this scenario. We have put this into our backlog and shall get back to you incase we need more input.

**Reply by dwelch2344** [reply]
@monishcm any word on where this is in the backlog? This data would be a HUGE win to the entire community. 

Our company would be happy to provide PRs, and/or collaborate + host the data (openly for everyone) long term. The evolution and adoption of packages over timelines would be uber useful for a number of cases.

Other threads where this has been discussed: 
- https://github.com/npm/feedback/discussions/306
